test/Microsoft.Extensions.Logging.EventSource.Test/EventSourceLoggerTest.cs

```diff
@@ -0,0 +1,581 @@
+
+                logger2.LogCritical(new EventId(5), new Exception("oops", new Exception("inner oops")),
+                    "Logger2 Event5 Critical {stringParam} {int1Param} {int2Param}", "bar", 23, 45);
+
+                using (logger3.BeginScope("Inner scope {timeParam} {guidParam}", TimeParam, GuidParam))
+                {
+                    logger2.LogWarning(new EventId(6), "Logger2 Event6 Warning NoParams");
+                }
+
+                logger3.LogInformation(new EventId(7), "Logger3 Event7 Information {stringParam} {doubleParam} {intParam}", "inner scope closed", DoubleParam2, 37);
+            }
+
+            logger2.LogWarning(new EventId(8), "Logger2 Event8 Warning {stringParam} {timeParam}", "Outer scope closed", TimeParam.ToString("O"));
+        }
+
+        private static void VerifyEvents(TestEventListener eventListener, params string[] verifierIDs)
+        {
+            Assert.Collection(eventListener.Events, verifierIDs.Select(id => EventVerifiers[id]).ToArray());
+        }
+
+        private static void VerifySingleEvent(string eventJson, string loggerName, string eventName, int? eventId, LogLevel? level, params string[] fragments)
+        {
+            Assert.True(eventJson.Contains(@"""__EVENT_NAME"":""" + eventName + @""""), $"Event name does not match. Expected {eventName}, event data is '{eventJson}'");
+            Assert.True(eventJson.Contains(@"""LoggerName"":""" + loggerName + @""""), $"Logger name does not match. Expected {loggerName}, event data is '{eventJson}'");
+            if (level.HasValue)
+            {
+                Assert.True(eventJson.Contains(@"""Level"":" + ((int)level.Value).ToString()), $"Log level does not match. Expected level {((int)level.Value).ToString()}, event data is '{eventJson}'");
+            }
+            if (eventId.HasValue)
+            {
+                Assert.True(eventJson.Contains(@"""EventId"":""" + eventId.Value.ToString()), $"Event id does not match. Expected id {eventId.Value}, event data is '{eventJson}'");
+            }
+
+            for (int i=0; i<fragments.Length; i++)
+            {
+                Assert.True(eventJson.Contains(fragments[i]), $"Event data '{eventJson}' does not contain expected fragment {fragments[i]}");
+            }
+        }
+
+        private class TestEventListener : EventListener
+        {
+            // This is a workaround for a bug in EventListener where OnEventSourceCreated will be called before the listener
+            // is fully constructed. Unfortunately this bug has shipped and is now an "expected behavior".
+            public class ListenerSettings
+            {
+                public EventKeywords Keywords;
+                public EventLevel Level;
+                public string FilterSpec;
+            }
+            public static ListenerSettings Settings = new ListenerSettings();
```

Comment:
I have measured--the execution of these 11 tests takes less than 300 ms, and that includes Console.WriteLine calls to report the timing. I suggest we leave them as they are.